numpy/core/src/multiarray/mapping.c

```diff
@@ -1446,23 +1467,23 @@ array_subscript(PyArrayObject *self, PyObject *op)
     PyArrayObject *view = NULL;
     PyObject *result = NULL;
 
-    PyArrayMapIterObject * mit = NULL;
+    PyArrayMapIterObject *mit = NULL;
 
-    /* return fields if op is a string index */
+    /*
+     * If the provided arg is a string, it might be attempting to index
+     * a field in a structured array.
+     */
     if (PyDataType_HASFIELDS(PyArray_DESCR(self))) {
-        PyArrayObject *view;
```

Comment:
I'd argue it was better to leave this here, as it indicates to the reader that the outer `view` is not changed by this branch.

Reply:
the outer `view` **is** changed, though, no?

Reply:
No, not by this code path. This `view` is scoped only to this `if` statement.

Reply:
If we redefine `PyArrayObject *view;` within the `if` block, I agree that it will shadow the outer `PyArrayObject *view;`.

But it's not clear to me why this is advantageous. Isn't it preferable to allocate only a single pointer?

Reply:
I would trust the compiler to get it right, the only thing would be to silence static analyzer warnings about it.  But in that case, we could call it `field_view` here and get no shadowing and clarity that `view` is not set.